### DISCUSSION TITLE
[Config Support]: Playback uses hevc and fails, rather than h264 as expected

### DISCUSSION THREAD

**Discussion by mileshampson** [blakeblackshear/frigate#13881] [Config Support] [answered]
### Describe the problem you are having

I have tried to configure frigate to use a _sub stream (which I have confirmed with ffprobe uses h264 rather than hevc) for detection, and the impression I got from the docs was that this would control live playback of clips in the browser (I have also tried setting the live: stream name parameter to ensure this). However, while the preview I get from the timeline scrubber works, any attempt to play a clip from the timeline yields the error:
```
Failed to play recordings (error 3): PIPELINE_ERROR_DECODE: VaapiVideoDecoder: failed Initialize()ing the frame pool
```
(or if I turn off "use graphics acceleration when available" in my Chromium 129 it just hangs). chrome://media-internals/ has:
```
00:00:00.000	created	"2024-09-22 08:31:08.719525 UTC"
00:00:00.000	origin_url	"http://localhost:5000/"
00:00:00.000	kFrameUrl	"http://localhost:5000/review"
00:00:00.000	kFrameTitle	"Recordings - Frigate"
00:00:00.000	url	"blob:http://localhost:5000/18720a67-2727-4a71-a71a-f60ae7fa0204"
00:00:00.000	info	"ChunkDemuxer"
00:00:00.000	kRendererName	"RendererImpl"
00:00:00.000	pipeline_state	"kStarting"
00:00:00.526	duration	1859.237999
00:00:01.128	kVideoTracks	[{"alpha mode":"is_opaque","codec":"hevc","coded size":"4512x2512","color space":{"matrix":"BT709","primaries":"BT709","range":"LIMITED","transfer":"BT709"},"encryption scheme":"Unencrypted","has extra data":false,"hdr metadata":"unset","natural size":"4512x2512","orientation":"0°","profile":"hevc main","visible rect":"0,0 4512x2512"}]
00:00:01.128	info	"Cannot select DecryptingVideoDecoder for video decoding"
00:00:01.137	info	"VideoDecoderPipeline |decoder_| Initialize() successful"
00:00:01.137	kIsVideoDecryptingDemuxerStream	false
00:00:01.137	kVideoDecoderName	"VaapiVideoDecoder"
00:00:01.137	kIsPlatformVideoDecoder	true
00:00:01.137	info	"Selected VaapiVideoDecoder for video decoding, config: codec: hevc, profile: hevc main, level: not available, alpha_mode: is_opaque, coded size: [4512,2512], visible rect: [0,0,4512,2512], natural size: [4512,2512], has extra data: false, encryption scheme: Unencrypted, rotation: 0°, flipped: 0, color space: {primaries:BT709, transfer:BT709, matrix:BT709, range:LIMITED}"
00:00:01.137	pipeline_state	"kPlaying"
00:00:01.147	error	"VaapiVideoDecoder: failed Initialize()ing the frame pool"
00:00:01.147	warning	"video decoder fallback after initial decode error."
00:00:01.147	info	"Cannot select VpxVideoDecoder for video decoding"
00:00:01.148	info	"Cannot select Dav1dVideoDecoder for video decoding"
00:00:01.148	info	"Cannot select FFmpegVideoDecoder for video decoding"
00:00:01.148	error	{"cause":{"code":1,"data":{},"group":"DecoderStatus","message":"","stack":[{"file":"media/gpu/vaapi/vaapi_video_decoder.cc","line":1217}]},"code":3,"data":{},"group":"PipelineStatus","message":"","stack":[{"file":"media/renderers/video_renderer_impl.cc","line":592}]}
```
which mentions hevc so looks to me like it is still trying to use my main, rather than sub, stream. In my included config I have a go2rtc setup, I have also tried without this. I have also tried without input_args: preset-rtsp-restream
The only thing that works is changing my recording to use the h264Preview_01_sub as well, which I don't want to do since the h264Preview_01_main is much higher resolution. Have I misconfigured something or misunderstood?

### Version

0.14.1-f4f3cfa

### Frigate config file

```yaml
mqtt:
  enabled: false
detectors:
  coral:
    type: edgetpu
    device: usb
detect:
  enabled: True
  width: 896
  height: 512
  fps: 5
ffmpeg:
  hwaccel_args: preset-vaapi
record:
  enabled: True
  retain:
    days: N
    mode: all
live:
  stream_name: h264Preview_01_sub
go2rtc:
  streams:
    2_Front_Door:
      - rtsp://user:pass@N.N.N.N1:554/h264Preview_01_sub
    3_Front_Right_Garage:
      - rtsp://user:pass@N.N.N.N2:554/h264Preview_01_sub
cameras:
  2_Front_Door:
    ffmpeg:
      inputs:
        - path: rtsp://user:pass@N.N.N.N1:554/h264Preview_01_main
          roles:
            - record
        - path: rtsp://user:pass@N.N.N.N1:554/h264Preview_01_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
  3_Front_Right_Garage:
    ffmpeg:
      inputs:
        - path: rtsp://user:pass@N.N.N.N2:554/h264Preview_01_main
          roles:
            - record
        - path: rtsp://user:pass@N.N.N.N2:554/h264Preview_01_sub
          input_args: preset-rtsp-restream
          roles:
            - detect
version: 0.14
```

### Relevant Frigate log output

```shell
No errors
```

### Relevant go2rtc log output

```shell
No errors
```

### Operating system

Other Linux

### Install method

Docker Compose

### Object Detector

Coral

**Comment by hawkeye217** [comment]
The `stream_name` directive is only applicable to Frigate's Live view. Any of your recorded footage is not processed by Frigate (unless explicitly transcoded, for instance, by go2rtc), so if you need your recorded footage to be h.264, you'll need to change the output in your camera's firmware or transcode using go2rtc, eg:

```
go2rtc:
  streams:
    2_Front_Door:
      - rtsp://user:pass@N.N.N.N1:554/h264Preview_01_sub
    2_Front_Door_Main:
      - rtsp://user:pass@N.N.N.N1:554/h264Preview_01_main#video=h264#audio=copy#hardware
    3_Front_Right_Garage:
      - rtsp://user:pass@N.N.N.N2:554/h264Preview_01_sub
cameras:
  2_Front_Door:
    ffmpeg:
      inputs:
        - path: rtsp://127.0.0.1:8554/2_Front_Door_Main
          roles:
            - record
 ...
```

Additionally, if you do plan to use `stream_name`, you should specify that on the camera level, not at the global level.

**Reply by mileshampson** [reply]
Thank you, that helped me understand that the review panel is using the recorded video on disk (which in my case is x265) to play back clips, not the x264 substream I have configured for detect. I assume this is to avoid recording multiple streams to disk, and there is no way to set it up to use a different sub stream to the recorded video.
In that case is there any way to view hevc video in the review panel rather than transcoding my recordings to x264 or using the lower quality sub stream for recording?

**Reply by NickM-27** [reply]
use a browser that supports hevc

**Reply by hawkeye217** [reply, marked as answer]
Not all browsers support all codecs, so you'll need to use a browser that supports the codec you want to use. Unfortunately copyrights, patents, and profit drives a lot of the support. See this chart in the go2rtc readme for codec support: https://github.com/AlexxIT/go2rtc/#codecs-madness

**Reply by mileshampson** [reply]
That is wild, I had no idea it was like that. Following that link I found out that fortunately in my case I just needed to pass `--enable-features=PlatformHEVCEncoderSupport` to chromium v129 to get it working. Thanks for your help!

**Reply by hawkeye217** [reply]
Great! Happy to help.
